src/click/types.py

```diff
@@ -297,16 +297,15 @@ def convert(
         if normed_value in normed_choices:
             return normed_choices[normed_value]
 
+        self.fail(self.get_missing_message(value), param, ctx)
```

Comment:
This does not call the new function you created.

Reply:
Resolved!